tests/terratest/functions/set/setGKE.go

```diff
@@ -0,0 +1,104 @@
+	config.LoadConfig("rancher", rancherConfig)
+
+	terraformConfig := new(terratest.TerraformConfig)
+	config.LoadConfig("terraform", terraformConfig)
+
+	googleAuthEncodedJSONConfig := new(terratest.GoogleAuthEncodedJSON)
+	config.LoadConfig("googleAuthEncodedJSON", googleAuthEncodedJSONConfig)
+
+	// create new empty hcl file object
+	f := hclwrite.NewEmptyFile()
+
+	// initialize the body of the new file object
+	rootBody := f.Body()
+
+	// initialize terraform object and set req provider version
+	tfBlock := rootBody.AppendNewBlock("terraform", nil)
+	tfBlockBody := tfBlock.Body()
+	reqProvsBlock := tfBlockBody.AppendNewBlock("required_providers", nil)
+	reqProvsBlockBody := reqProvsBlock.Body()
+
+	reqProvsBlockBody.SetAttributeValue("rancher2", cty.ObjectVal(map[string]cty.Value{
+		"source":  cty.StringVal("rancher/rancher2"),
+		"version": cty.StringVal(terraformConfig.ProviderVersion),
+	}))
+	rootBody.AppendNewline()
+
+	// Provider
+	provBlock := rootBody.AppendNewBlock("provider", []string{"rancher2"})
+	provBlockBody := provBlock.Body()
+	provBlockBody.SetAttributeValue("api_url", cty.StringVal(`https://`+rancherConfig.Host))
+	provBlockBody.SetAttributeValue("token_key", cty.StringVal(rancherConfig.AdminToken))
+	provBlockBody.SetAttributeValue("insecure", cty.BoolVal(*rancherConfig.Insecure))
+	rootBody.AppendNewline()
+
+	// Resource cloud credential
+	cloudCredBlock := rootBody.AppendNewBlock("resource", []string{"rancher2_cloud_credential", "rancher2_cloud_credential"})
+	cloudCredBlockBody := cloudCredBlock.Body()
+	cloudCredBlockBody.SetAttributeValue("name", cty.StringVal(terraformConfig.CloudCredentialName))
+	googleCredConfigBlock := cloudCredBlockBody.AppendNewBlock("google_credential_config", nil)
+	authEncodedJSON := hclwrite.Tokens{
+		{Type: hclsyntax.TokenIdent, Bytes: []byte(`jsonencode({"type" = "` + googleAuthEncodedJSONConfig.Type + `", "project_id" = "` + googleAuthEncodedJSONConfig.ProjectID + `", "private_key_id" = "` + googleAuthEncodedJSONConfig.PrivateKeyID + `", "private_key" = "` + googleAuthEncodedJSONConfig.PrivateKey + `", "client_email" = "` + googleAuthEncodedJSONConfig.ClientEmail + `", "client_id" = "` + googleAuthEncodedJSONConfig.ClientID + `", "auth_uri" = "` + googleAuthEncodedJSONConfig.AuthURI + `", "token_uri" = "` + googleAuthEncodedJSONConfig.TokenURI + `", "auth_provider_x509_cert_url" = "` + googleAuthEncodedJSONConfig.AuthProviderX509CertURL + `", "client_x509_cert_url" = "` + googleAuthEncodedJSONConfig.ClientX509CertURL + `"})`)},
+	}
+	googleCredConfigBlock.Body().SetAttributeRaw("auth_encoded_json", authEncodedJSON)
+	rootBody.AppendNewline()
+
+	// Resource cluster
+	clusterBlock := rootBody.AppendNewBlock("resource", []string{"rancher2_cluster", "rancher2_cluster"})
+	clusterBlockBody := clusterBlock.Body()
+	clusterBlockBody.SetAttributeValue("name", cty.StringVal(terraformConfig.ClusterName))
+	gkeConfigBlock := clusterBlockBody.AppendNewBlock("gke_config_v2", nil)
```

Comment:
Same nit here about code readability and separating variables out

Reply:
Are you talking about formatting and spacing? What do you mean separating variables out?

Reply:
Correct, just formatting and spacing.